backends/bmv2/common/action.cpp

```diff
@@ -70,9 +70,14 @@ void ActionConverter::convertActionBody(const IR::Vector<IR::StatOrDecl>* body,
             auto primitive = mkPrimitive(operation, result);
             auto parameters = mkParameters(primitive);
             primitive->emplace_non_null("source_info", assign->sourceInfoJsonObj());
-            auto left = ctxt->conv->convertLeftValue(l);
-            parameters->append(left);
             bool convertBool = type->is<IR::Type_Boolean>();
+            Util::IJson* left;
+            if (ctxt->conv->IsArrayIndexRuntime(l)) {
+                left = ctxt->conv->convert(l, true, true, convertBool);
```

Comment:
It is needed to deal with the case when a runtime index is a left value and is the only expression in JSON (without any other arithmetic operations).  The JSON format needs an extra wrapping `"type": "expression"`.

Regarding convertLeftValue, note, the right value is also using `convert` because simple_switch supports runtime index in right value with simple_switch.  See this code few lines below the left code.

`auto right = ctxt->conv->convert(r, true, true, convertBool);`